solidity/contracts/domains/noto/Noto.sol

```diff
@@ -253,4 +283,106 @@ contract Noto is EIP712Upgradeable, UUPSUpgradeable, INoto {
         );
         return _hashTypedDataV4(structHash);
     }
+
+    /**
+     * @dev Create a new locked state that can only be unlocked by a specific delegate.
+     *
+     * @param locked new output state to generate, representing locked value
+     * @param lock details on the lock and any possible outcomes
+     * @param signature EIP-712 signature on the original request that spawned this transaction
+     * @param data any additional transaction data (opaque to the blockchain)
+     */
+    function createLock(
+        bytes32 locked,
+        LockInput calldata lock,
+        bytes calldata signature,
+        bytes calldata data
+    ) public virtual override onlyNotary {
+        LockDetail storage stored = _locks[locked];
+        for (uint256 i = 0; i < lock.outcomes.length; i++) {
+            stored.outcomes[lock.outcomes[i].ref] = lock.outcomes[i].state;
+        }
+        stored.delegate = lock.delegate;
+        stored.initialized = true;
+        stored.data = data;
+        emit NotoLock(locked, signature, data);
+    }
+
+    /**
+     * @dev Perform a transfer and a lock simultaneously.
+     *
+     * @param inputs as per transfer()
+     * @param unlockedOutputs as per transfer()
+     * @param lockedOutput as per createLock()
+     * @param lock as per createLock()
+     * @param signature as per createLock()
+     * @param data as per createLock()
+     */
+    function transferAndLock(
+        bytes32[] calldata inputs,
+        bytes32[] calldata unlockedOutputs,
+        bytes32 lockedOutput,
+        LockInput calldata lock,
+        bytes calldata signature,
+        bytes calldata data
+    ) external virtual override onlyNotary {
+        _transfer(inputs, unlockedOutputs, signature, data);
```

Comment:
feel like it'd be more intuitive to use separate signatures for the transfer() vs. createLock() authorizations